components/cldr-json-data-provider/src/transform/dates.rs

```diff
@@ -0,0 +1,368 @@
+use crate::cldr_langid::CldrLangID;
+use crate::error::Error;
+use crate::reader::{get_subdirectories, open_reader};
+use crate::support::DataKeySupport;
+use crate::CldrPaths;
+use icu_data_provider::iter::DataEntryCollection;
+use icu_data_provider::prelude::*;
+use icu_data_provider::structs::dates::*;
+use std::convert::TryFrom;
+use std::marker::PhantomData;
+
+/// A data provider reading from CLDR JSON dates files.
+#[derive(PartialEq, Debug)]
+pub struct DatesProvider<'d> {
+    data: Vec<(CldrLangID, cldr_json::LangDates)>,
```

Comment:
Suggestion: make this type `cldr_json::LangData`.  If you do this, then you can concatenate each locale's `cldr_json::LangData` to the main copy and simplify the `try_from` function.  Something like this:

```rust
        for dir in locale_dirs {
            let path = dir.join("ca-gregorian.json");
            let resource: cldr_json::Resource =
                serde_json::from_reader(open_reader(path)?)?;
            data.append(resource.main.0);
        }
```

And if it makes you feel better, after you complete the loop, you can double-check that `data` is indeed sorted.

Reply:
I'd prefer not to assume order returned by `read_dir`

Reply:
Consider pre-sorting `locale_dirs`.

Reply:
sorted!